lib/utils.js

```diff
@@ -1,52 +1,122 @@
+      return Winston.error('Fetching facility data failed', {err: err.message})
     }
 
     if (res.statusCode !== 200) {
-      return Winston.error('Non-200 status code: %d', res.statusCode);
+      return Winston.error('Non-200 status code: %d', res.statusCode)
     }
 
-    var title = res.body.title;
-    var headers = res.body.headers;
+    var title = res.body.title
+    var headers = res.body.headers
 
-    var batch = Cache.batch();
+    var batch = Cache.batch()
 
     // Add each facility to the cache
-    res.body.rows.forEach(function(row) {
-      var key = row[0];
+    res.body.rows.forEach(function (row) {
+      var key = row[0]
       if (!key) {
-        return Winston.error('Facility missing value', {facility: row});
+        return Winston.error('Facility missing value', {facility: row})
       }
       var body = Mustache.render(template, {
         title: title,
         headers: JSON.stringify(headers),
         row: JSON.stringify(row),
         width: headers.length
-      });
-      batch.put(key, body);
-    });
+      })
+      batch.put(key, body)
+    })
 
-    batch.write(function(err) {
+    batch.write(function (err) {
       if (err) {
-        return Winston.error('Storing facility data failed', {err: err.message});
+        return Winston.error('Storing facility data failed', {err: err.message})
       }
-      Winston.info('Finished populating cache');
-    });
-  });
-};
+      Winston.info('Finished populating cache')
+    })
+  })
+}
+
+let cronJobs = {}
+
+let addCronJob = exports.addCronJob = (pattern, timezone, url) => {
+  cronJobs[url] = new CronJob(pattern, populateCache.bind(null, url), true, timezone)
+}
+
+let removeCronJob = (url) => {
+  if (cronJobs[url]) {
+    cronJobs[url].stop()
+    delete cronJobs[url]
+  }
+}
+
+let addRoute = exports.addRoute = (app, path) => {
+  app.get(path, Routes.facilityCodeLookup)
+}
+
+let removeCacheForRoute = (app, path) => {
+  removeRoute(app, path)
+}
+
+exports.updateConfig = (app, oldConf, newConf) => {
+  let newRoutes = {}
+  let existingRoutes = {}
+
+  newConf.forEach((route) => {
+    var url = URL.resolve(route.dhisUrl, route.dhisPath)
+    if (!cronJobs[url]) {
+      newRoutes[url] = {
+        host: route.dhisUrl,
+        path: route.dhisPath,
+        pattern: route.cronPattern,
+        timezone: route.cronTimezone
+      }
+    } else {
+      existingRoutes[url] = {
+        pattern: route.cronPattern,
+        timezone: route.cronTimezone
+      }
+    }
+  })
+
+  oldConf.forEach((route) => {
+    var url = URL.resolve(route.dhisUrl, route.dhisPath)
+    if (!newRoutes[url] && !existingRoutes[url]) {
+      // delete old routes and crons
+      removeCacheForRoute(app, route.dhisPath)
+    }
+    removeCronJob(url)
+  })
+
+  for (var j in newRoutes) {
+    addRoute(app, newRoutes[j].path)
+    addCronJob(newRoutes[j].pattern, newRoutes[j].timezone, j)
+    populateCache(j)
+  }
+
+  for (var k in existingRoutes) {
```

Comment:
Can't we get rid of this loop by adding the cron job in an else block of the condition on [line 102](https://github.com/jembi/facility-cache/pull/1/files#diff-50e3aa130a4f97a42ee2cf111c7b1d9dR102)? The removing of the cron job can be move to above that condition.